tests/Decoder.js

```diff
@@ -275,13 +275,13 @@ test('Decode optional arguments', t => {
     t.plan(2)
     t.deepEqual(
         encoder.decode(CONTRACT, 'test_optional', 'cb_r4IAAQA/aHG2bw=='),
-        {None: []},
```

Comment:
The implementation should accept the generic format as well, so please keep this test cases.

Reply:
Should we define an option to switch between decoding to `undefined` or `{ None: [] }`?

Reply:
My bad, i overlooked this the the decoder. However, I prefer `null` than `undefined` (it's OK to accept it in the encoder tho)

Reply:
I took the TypeScript semantic, they have optional arguments like `Number?` that is defined as `Number | undefined`. So, I assume that the optional return value should be defined the same, to make possible to
```
function bar(): number | undefined;
function foo(t: number?);

foo(bar());
```
instead of
```
function bar(): number | null;
function foo(t: number | null); // can't use syntax for optional arguments

foo(bar());
```
or even
```
function bar(): number | null;
function foo(t: number?);

const t = bar()
foo(t === null ? undefined : t);
```

Reply:
okay, let it be TypeScript semantics then